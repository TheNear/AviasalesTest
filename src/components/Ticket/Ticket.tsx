import React from "react";
import TicketInfo from "../../componentsHelper/TicketInfo/TicketInfo";
import style from "./Ticket.module.css";

const Ticket: React.FC = () => {
  return (
    <li className={style.ticket}>
      <div className={style.price}>13 400 P</div>
      <img
        className={style.image}
        src="http://pics.avs.io/99/36/S7.png"
        alt="airport"
      />
      <TicketInfo />
      <TicketInfo />
      <TicketInfo />
      <TicketInfo />
      <TicketInfo />
      <TicketInfo />
    </li>
  );
};

export default Ticket;
