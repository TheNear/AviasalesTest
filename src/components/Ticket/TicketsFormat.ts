export const stopsFormat: Array<string> = [
  "Без пересадок",
  "1 Пересадка",
  "2 Пересадки",
  "3 Пересадки",
  "4 Пересадки",
  "5 Пересадок",
  "6 Пересадок",
  "7 Пересадок",
  "8 Пересадок",
  "9 Пересадок",
  "10 Пересадок",
];

export const convertPrice = (number: number): string => {
  return new Intl.NumberFormat("ru-RU").format(number);
};

export const getTravelTime = (time: number): string => {
  const dateObj = new Date(time * 1000 * 60);
  const hours = dateObj.getHours();
  const minutes = dateObj.getMinutes();
  return `${hours}ч ${minutes}м`;
}