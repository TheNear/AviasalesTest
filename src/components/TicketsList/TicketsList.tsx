import React, { useCallback, useEffect } from "react";
import { useDispatch } from "react-redux";
import style from "./TicketsList.module.css";
import { Loader } from "../../componentsHelper/Loader/Loader";
import { Error } from "../../componentsHelper/Error/Error";
import { Ticket } from "../Ticket/Ticket";
import { ITicketItem } from "../../types/tickets";
import { getTickets } from "../../redux/tickets/action";
import { useTickets } from "../../hooks/useTickets";

// interface ITicketsList {
//   getQueries: GetKeysValuesType;
// }

const TicketsList: React.FC = () => {
  const dispatch = useDispatch();
  const { tickets, error, isFetching } = useTickets();

  const getTicketsList = useCallback(() => {
    dispatch(getTickets());
  }, [dispatch]);

  useEffect(getTicketsList, [getTicketsList]);

  return (
    <ul className={style.tickets}>
      {isFetching && <Loader />}
      {!isFetching && error && <Error refreshTicketsList={getTicketsList} />}
      {tickets.map((ticket: ITicketItem) => (
        <Ticket
          key={`${ticket.price}${ticket.segments[0].date}${ticket.carrier}`}
          ticket={ticket}
        />
      ))}
    </ul>
  );
};

export default TicketsList;
