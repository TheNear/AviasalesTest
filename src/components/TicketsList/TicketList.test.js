import React from "react";
import { shallow } from "enzyme";
import TicketsList from "./TicketsList";
import { ticketMock } from "../Ticket/ticketsMock";

describe("Test TicketsList component", () => {
  const ticketMockList = [ticketMock, ticketMock, ticketMock];

  it("Render empty TicketsList", () => {
    const wrapper = shallow(<TicketsList tickets={ticketMockList} />);
    expect(wrapper).toMatchSnapshot();
  });
});
