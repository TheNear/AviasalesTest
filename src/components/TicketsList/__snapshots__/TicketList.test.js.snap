// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TicketsList component Render empty TicketsList 1`] = `
<ul
  className="tickets"
>
  <Ticket
    key="182042020-10-13T13:46:00.000ZS7"
    ticket={
      Object {
        "carrier": "S7",
        "price": 18204,
        "segments": Array [
          Object {
            "date": "2020-10-13T13:46:00.000Z",
            "destination": "HKT",
            "duration": 1544,
            "origin": "HKT",
            "stops": Array [
              "SHA",
              "DXB",
              "HKG",
            ],
          },
          Object {
            "date": "2020-10-13T13:46:00.000Z",
            "destination": "HKT",
            "duration": 1944,
            "origin": "HKT",
            "stops": Array [
              "SHA",
              "DXB",
              "HKG",
            ],
          },
        ],
      }
    }
  />
  <Ticket
    key="182042020-10-13T13:46:00.000ZS7"
    ticket={
      Object {
        "carrier": "S7",
        "price": 18204,
        "segments": Array [
          Object {
            "date": "2020-10-13T13:46:00.000Z",
            "destination": "HKT",
            "duration": 1544,
            "origin": "HKT",
            "stops": Array [
              "SHA",
              "DXB",
              "HKG",
            ],
          },
          Object {
            "date": "2020-10-13T13:46:00.000Z",
            "destination": "HKT",
            "duration": 1944,
            "origin": "HKT",
            "stops": Array [
              "SHA",
              "DXB",
              "HKG",
            ],
          },
        ],
      }
    }
  />
  <Ticket
    key="182042020-10-13T13:46:00.000ZS7"
    ticket={
      Object {
        "carrier": "S7",
        "price": 18204,
        "segments": Array [
          Object {
            "date": "2020-10-13T13:46:00.000Z",
            "destination": "HKT",
            "duration": 1544,
            "origin": "HKT",
            "stops": Array [
              "SHA",
              "DXB",
              "HKG",
            ],
          },
          Object {
            "date": "2020-10-13T13:46:00.000Z",
            "destination": "HKT",
            "duration": 1944,
            "origin": "HKT",
            "stops": Array [
              "SHA",
              "DXB",
              "HKG",
            ],
          },
        ],
      }
    }
  />
</ul>
`;
